flink-table/flink-table-planner/src/test/java/org/apache/flink/table/sqlexec/SqlToOperationConverterTest.java

```diff
@@ -163,6 +169,235 @@ public void testSqlInsertWithStaticPartition() {
+	}
+
+	@Test
+	public void testCreateTableWithUnSupportedDataType1() {
+		expectedEx.expect(TableException.class);
+		expectedEx.expectMessage("Type is not supported: TIMESTAMP_WITH_LOCAL_TIME_ZONE");
+		final String sql = "create table t1(\n" +
+			"  f41 ARRAY<TIMESTAMP(3) WITH LOCAL TIME ZONE>)";
+		final FlinkPlannerImpl planner = getPlannerBySqlDialect(SqlDialect.DEFAULT);
+		SqlNode node = planner.parse(sql);
+		assert node instanceof SqlCreateTable;
+		SqlToOperationConverter.convert(planner, node);
+	}
+
+	@Test
+	public void testCreateTableWithUnSupportedDataType2() {
+		expectedEx.expect(TableException.class);
+		expectedEx.expectMessage("Type is not supported: TIMESTAMP_WITH_LOCAL_TIME_ZONE");
+		final String sql = "create table t1(\n" +
+			"  f40 TIMESTAMP(3) WITH LOCAL TIME ZONE)";
+		final FlinkPlannerImpl planner = getPlannerBySqlDialect(SqlDialect.DEFAULT);
+		SqlNode node = planner.parse(sql);
+		assert node instanceof SqlCreateTable;
+		SqlToOperationConverter.convert(planner, node);
+	}
+
+	@Test
+	public void testCreateTableWithUnSupportedDataType3() {
+		expectedEx.expect(TableException.class);
+		expectedEx.expectMessage("Type is not supported: TIMESTAMP_WITH_LOCAL_TIME_ZONE");
+		final String sql = "create table t1(\n" +
+			"  f39 TIMESTAMP WITH LOCAL TIME ZONE)";
+		final FlinkPlannerImpl planner = getPlannerBySqlDialect(SqlDialect.DEFAULT);
+		SqlNode node = planner.parse(sql);
+		assert node instanceof SqlCreateTable;
+		SqlToOperationConverter.convert(planner, node);
+	}
+
+	@Test
+	public void testCreateTableWithUnSupportedDataType4() {
+		expectedEx.expect(TableException.class);
+		expectedEx.expectMessage("Type is not supported: VARBINARY");
+		final String sql = "create table t1(\n" +
+			"  f12 BYTES)";
+		final FlinkPlannerImpl planner = getPlannerBySqlDialect(SqlDialect.DEFAULT);
+		SqlNode node = planner.parse(sql);
+		assert node instanceof SqlCreateTable;
+		SqlToOperationConverter.convert(planner, node);
+	}
+
+	@Test
+	public void testCreateTableWithUnSupportedDataType5() {
+		expectedEx.expect(TableException.class);
+		expectedEx.expectMessage("Type is not supported: VARBINARY");
+		final String sql = "create table t1(\n" +
+			"  f11 VARBINARY(33))";
+		final FlinkPlannerImpl planner = getPlannerBySqlDialect(SqlDialect.DEFAULT);
+		SqlNode node = planner.parse(sql);
+		assert node instanceof SqlCreateTable;
+		SqlToOperationConverter.convert(planner, node);
+	}
+
+	@Test
+	public void testCreateTableWithUnSupportedDataType6() {
+		expectedEx.expect(TableException.class);
+		expectedEx.expectMessage("Type is not supported: VARBINARY");
+		final String sql = "create table t1(\n" +
+			"  f10 VARBINARY)";
+		final FlinkPlannerImpl planner = getPlannerBySqlDialect(SqlDialect.DEFAULT);
+		SqlNode node = planner.parse(sql);
+		assert node instanceof SqlCreateTable;
+		SqlToOperationConverter.convert(planner, node);
+	}
+
+	@Test
+	public void testCreateTableWithUnSupportedDataType7() {
```

Comment:
Some feedback around readability of tests. We should avoid numbering tests and instead give them useful names e.g. instead of `testCreateTableWithUnSupportedDataType7` why not `testCreateTableUnsupportedBinary`? And same concern as before, a test spec would reduce code duplication. Take `org.apache.flink.table.types.LogicalTypeParserTest` as a reference for this class and future tests.